user/TASK/gimbal_task/gimbal_task.c

```diff
@@ -29,21 +28,43 @@
 #include <stdio.h>
 #include "pid.h"
 #include "shoot_task.h"
+#include <math.h>
 
-#define DEADBAND 10
+#define DEADBAND 1
 
 // This is accessbile globally and some data is loaded from INS_task
 Gimbal_t gimbal;
 
-//UART mailbox
-char str[32] = {0};
+static int loop_counter = 0;
 
 /******************** Functions ********************/
 static void initialization(Gimbal_t *gimbal);
 static void get_new_data(Gimbal_t *gimbal);
 static void update_setpoints(Gimbal_t *gimbal);
 static void increment_PID(Gimbal_t *gimbal);
+static void fill_complex_equivalent(fp32 position[2], uint16_t ecd_value);
 
+/**
+* Slew Control
+* While the current is over a limiting constant:
+* Reduce control signal by 0.5 every 5 ms
+* When current returns the acceptable range, double the 
+* allowed current every 5 ms until it returns to full output. 
+* 
+* Disable/reset I-term at this time whenever current limiting is active.\
+* 
+* Do we want reductions to follow : 1/2, 1/4, 1/8, ... 
+*                                    or 1/2, 1/4 , OFF
+* Or mutliply out put constant
+* Tunable Parameters
+* redutions factor: e.g. reduce current by 0.5
+* reduction time: e.g. reduce current every 5 ms
+* increase factor
+* increase time
+* I-term off/reset
+* hysteresis time: how long should the current limiter 
+*       wait before starting to increase/decrease current.
+*/
```

Comment:
Should be a TODO at the top of this - a larger, "project level" TODO like this seems ok to push

Reply:
deleted